.circleci/config.yml

```diff
@@ -432,6 +432,7 @@ workflows:
             branches:
               ignore:
                 - /release-[0-9]+\.[0-9]+.*/ # match with releaseX.Y.*
+                - master
```

Comment:
I don't think we want to skip `check-merge-to-enterprise` job on `master`, because when our PR fails on this job, we want to know if that is related to our PR, or if `master` also fails on this job. If `master` fails on this job, it means that we should first fix the master. This job is not required so it is possible for `master` to fail on this job. Overall I find value in having runs on `master`.

Reply:
I think that we have too many false negatives for this test. For example, the test fails on master with the error message:
```
ERROR: enterprise/master was not found and community PR branch could not be merged into enterprise-master
```

It is not logical to search for a branch named `master` on `enterprise` repository.


However, we can always improve `check_enterprise_merge.sh` to have more custom logic for master branch, and perform a different set of steps to actually make sure everything is ok

Reply:
I think the failure is not a false negative because if there was no problem in `check merge to enterprise` then we would not see this error. If there are no conflicts, the script will return here:
https://github.com/citusdata/citus/blob/acdde9fa2de2cda749d8b1654c52734148a93401/ci/check_enterprise_merge.sh#L58-L60

If there are conflicts then we try to see if there is a branch named the same in enterprise, and in case of `master` branch that surely doesn't exist in enterprise. Actually we don't need to continue after the above block if `master` branch fails. I don't think we need to add a custom logic to our `check_enterprise.merge.sh` for this specific case but if you think it is confusing, you could return early in the script if the branch name is `master` and we fail the above if block.

If we look at the example output you shared, we can see that there are actual conflicts:

![image](https://user-images.githubusercontent.com/17179875/117956134-6da8be00-b321-11eb-821a-896100133f3f.png)

Reply:
I realized that the issue is the following:

- If we open a PR on community that breaks automatic merges to enterprise, we open a sister PR on enterprise.
- After we merge the community branch, and before the sister branch is merged to enterprise, we are in a state where we can not automatically merge community/master into enterprise-master.

This is totally expected and in fact this is the reason why we opened that sister PR on enterprise repository in the first place. However CI does not really know about it.

After we merge the PR on enterprise and restart the failing test on master, it starts to pass and we have a green light again.

My solution is this: Merge the PR on enterprise first, and community second. This way the tests will be green on master. Does this make sense @SaitTalhaNisanci ?

Reply:
It might make sense to update the readme, but I don't think anyone will actually notice the difference.

```diff
diff --git a/ci/README.md b/ci/README.md
index 52a8925cc..6e221b351 100644
--- a/ci/README.md
+++ b/ci/README.md
@@ -156,9 +156,9 @@ git merge "community/$PR_BRANCH"
    familiar with the change.
 5. You should rerun the `check-merge-to-enterprise` check on
    `community/$PR_BRANCH`. You can use re-run from failed option in circle CI.
-6. You can now merge the PR on community. Be sure to NOT use "squash and merge",
+6. You can now merge the PR on enterprise. Be sure to NOT use "squash and merge",
    but instead use the regular "merge commit" mode.
-7. You can now merge the PR on enterprise. Be sure to NOT use "squash and merge",
+7. You can now merge the PR on community. Be sure to NOT use "squash and merge",
    but instead use the regular "merge commit" mode.

 The subsequent PRs on community will be able to pass the
```

Reply:
>My solution is this: Merge the PR on enterprise first, and community second. This way the tests will be green on master. Does this make sense @SaitTalhaNisanci ?

Actually I was doing it in this order for the exact reason you specified. We can update the readme in that way yeah :)